kayenta-graphite/src/main/java/com/netflix/kayenta/graphite/controller/GraphiteFetchController.java

```diff
@@ -0,0 +1,120 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.kayenta.graphite.controller;
+
+import static com.netflix.kayenta.canary.util.FetchControllerUtils.determineDefaultProperty;
+
+import com.netflix.kayenta.canary.CanaryMetricConfig;
+import com.netflix.kayenta.canary.CanaryScope;
+import com.netflix.kayenta.canary.providers.metrics.GraphiteCanaryMetricSetQueryConfig;
+import com.netflix.kayenta.graphite.config.GraphiteConfigurationTestControllerDefaultProperties;
+import com.netflix.kayenta.metrics.SynchronousQueryProcessor;
+import com.netflix.kayenta.security.AccountCredentials;
+import com.netflix.kayenta.security.AccountCredentialsRepository;
+import com.netflix.kayenta.security.CredentialsHelper;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.util.StringUtils;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.io.IOException;
+import java.time.Instant;
+import java.util.Collections;
+import java.util.Map;
+
+import io.swagger.annotations.ApiParam;
+import lombok.extern.slf4j.Slf4j;
+
+@RestController
+@RequestMapping("/fetch/graphite")
+@Slf4j
+public class GraphiteFetchController {
+    private final AccountCredentialsRepository accountCredentialsRepository;
+    private final SynchronousQueryProcessor synchronousQueryProcessor;
+    private final GraphiteConfigurationTestControllerDefaultProperties graphiteConfigurationTestControllerDefaultProperties;
+
+    @Autowired
+    public GraphiteFetchController(AccountCredentialsRepository accountCredentialsRepository,
+                                   SynchronousQueryProcessor synchronousQueryProcessor,
+                                   GraphiteConfigurationTestControllerDefaultProperties graphiteConfigurationTestControllerDefaultProperties) {
+        this.accountCredentialsRepository = accountCredentialsRepository;
+        this.synchronousQueryProcessor = synchronousQueryProcessor;
+        this.graphiteConfigurationTestControllerDefaultProperties =
+                graphiteConfigurationTestControllerDefaultProperties;
+    }
+
+    @RequestMapping(value = "/query", method = RequestMethod.POST)
+    public Map queryMetrics(@RequestParam(required = false) final String metricsAccountName,
+                            @RequestParam(required = false) final String storageAccountName,
+                            @ApiParam(defaultValue = "cpu") @RequestParam String metricSetName,
+                            @ApiParam(defaultValue = "system.cpu.user") @RequestParam String metricName,
```

Comment:
Seems like the `defaultValue` (or descriptive text) for `metricName` should give an example of using the templating that the metrics service's approach relies upon. That is, it should be clear to the user that they may want to include `$scope` or `$location` directly in `metricName`.

Reply:
thx, will add `$location` and `$scope` in default value and add some description in `location` and `scope`